src/mem/threadalloc.h

```diff
@@ -127,10 +127,11 @@ namespace snmalloc
      */
     static void inner_release()
     {
-      if (get() != &GlobalPlaceHolder)
+      Alloc*& a = get_noncachable();
```

Comment:
Should `get_noncacheable` return a reference?  It might be better to make it return a pointer and have a `get_reference` that is `private` and returns a reference to the `thread_local` thing.